pkg/account/account.go

```diff
@@ -0,0 +1,339 @@
+package account
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/Optum/dce/pkg/errors"
+	"github.com/Optum/dce/pkg/model"
+	validation "github.com/go-ozzo/ozzo-validation"
+)
+
+// Writer put an item into the data store
+type Writer interface {
+	WriteAccount(input *model.Account, lastModifiedOn *int64) error
+}
+
+// Deleter Deletes an Account from the data store
+type Deleter interface {
+	DeleteAccount(inptut *model.Account) error
+}
+
+// SingleReader Reads Account information from the data store
+type SingleReader interface {
+	GetAccountByID(accountID string, account *model.Account) error
+}
+
+// MultipleReader reads multiple accounts from the data store
+type MultipleReader interface {
+	GetAccounts(*model.Account) (*model.Accounts, error)
+	GetAccountsByStatus(status string) (*model.Accounts, error)
+}
+
+// Reader data Layer
+type Reader interface {
+	SingleReader
+	MultipleReader
+}
+
+// WriterDeleter data layer
+type WriterDeleter interface {
+	Writer
+	Deleter
+}
+
+// ReaderWriterDeleter includes Reader and Writer interfaces
+type ReaderWriterDeleter interface {
+	Reader
+	WriterDeleter
+}
+
+// Eventer for publishing events
+type Eventer interface {
+	Publish() error
+}
+
+// Manager manages all the actions against an account
+type Manager interface {
+	Setup(adminRole string) error
+}
+
+// Account is a type corresponding to a Account table record
+type Account struct {
+	writer WriterDeleter
```

Comment:
added this to the Account struct.  To @marinatedpork feedback I was passing this into a lot of functions and realized they are all the same.  The only thing that can be done to an account after this struct is instantiated is write/delete so just making it part of the struct instead of a parameter to the functions.  Notice this doesn't include the read commands.